tests/features/minter.feature

```diff
@@ -0,0 +1,64 @@
+Feature: Minter
+
+  Feature Description: Minter contract provides functionnalities to allow a user to buy project NFTs.
+  All corresponding actions must work as expected.
+
+  Background:
+    Given a deployed user contracts
+    And an admin with address 1000
+    And an anyone with address 1001
+    Given a deployed project nft contact
+    And owned by admin
+    Given a deployed payment token contact
+    And owned by admin
+    And a total supply set to 1,000,000
+    And anyone owns the whole supply
+    Given a deployed minter contact
+    And owned by admin
+    And a whitelist sale open
+    And a public sale close
+    And a max buy per tx set to 5
+    And an unit price set to 10
+    And a max supply set to 10
+    And a reserved supply set to 4
+    Given a whitelist merkle tree
+    And an allocation of 5 whitelist slots to anyone
+
+  Scenario: Whitelisted
```

Comment:
It's an open question: 
Do we want to expose all the scenariis on top of the test file (file header) or do we want to attached each scenario in each function header ?

Something like:
- Feature
- background

- scenario
- function

- scenario
- function

- scenario
- function


@Bal7hazar ?

Reply:
@cloudvenger,
Actually I did a duplication to have the information in the `.feature`  file and  in the test files spread at the most relevant places, background for setup the context and scenario into each E2E tests.

Spreading the scenarios helps to keep the link between expectations and tests.

I'm not a big fan of the duplication part, but as we said at the beginning we had to since there is no gherkin/cucumber deep management.